data/calculated_parser/parser.py

```
@@ -69,8 +70,43 @@ def get_key_for_variable(self, variable):
+        if not isinstance(key, tuple):
+            key = (key,)
 
+        d = dict(zip(self.dims, key))
+        try:
+            if hasattr(variable, "dims"):
+                # xarray calls it dims
+                key = [d[k] for k in variable.dims]
+            else:
+                key = [d[k] for k in variable.dimensions]
+        except KeyError:
+            raise SyntaxError
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)